<div style="text-align:left;font-size:2em"><span style="font-weight:bolder;font-size:1.25em">SP2273 | Learning Portfolio</span><br><br><span style="font-weight:bold;color:darkred">Using Jupyter (Good), Exercises</span></div>

# Exercise 1

In [1]:
while True:
    print('*', end='')

****************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************

KeyboardInterrupt: 

---------------------------------------------------------------------------
KeyboardInterrupt                         Traceback (most recent call last)
Cell In[1], line 2
      1 while True:
----> 2     print('*', end='')

File c:\Users\65881\miniconda3\miniconda\Lib\site-packages\IPython\core\interactiveshell.py:3053, in InteractiveShell._tee.<locals>.write(data, *args, **kwargs)
   3051 def write(data, *args, **kwargs):
   3052     """Write data to both the original destination and the capture dictionary."""
-> 3053     result = original_write(data, *args, **kwargs)
   3054     if any(
   3055         [
   3056             self.display_pub.is_publishing,
   (...)   3059         ]
   3060     ):
   3061         return result

File c:\Users\65881\miniconda3\miniconda\Lib\site-packages\ipykernel\iostream.py:676, in OutStream.write(self, string)
    673     msg = "I/O operation on closed file"
    674     raise ValueError(msg)
--> 676 is_child = not self._is_master_process()
    677 # only touch the buffer in the IO thread to avoid races
    678 with self._buffer_lock:

File c:\Users\65881\miniconda3\miniconda\Lib\site-packages\ipykernel\iostream.py:545, in OutStream._is_master_process(self)
...
--> 545 def _is_master_process(self):
    546     return os.getpid() == self._master_pid
    548 def set_parent(self, parent):

KeyboardInterrupt: 

# Exercise 2

In [7]:
print = 'PRINT'

In [3]:
print('Hello World')

TypeError: 'str' object is not callable

In [1]:
print('Hello World')

Hello World


# Exercise 3

In [ ]:
squares = [i**2 for i in range(10)]

: 

In [11]:
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [12]:
squares = list()
for i in range(10):
    squares.append(i**2)

In [13]:
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [14]:
squares = [i**2 for i in range(10)]
print(squares)

[0, 1, 4, 9, 16, 25, 36, 49, 64, 81]


In [7]:
%%timeit
squares = [i**2 for i in range(10)]

1.28 μs ± 88.1 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


The code took 1.28 microseconds to run once and the run time varied by about 88.1 nanoseconds. Python ran the code 7 separate times and each time looping 1 million times.

In [15]:
%%timeit
squares = list()
for i in range(10):
    squares.append(i**2)

1.5 μs ± 100 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)


The code took 1.5 microseconds to run once and the run time varied by about 100 nanoseconds. Python ran the code 7 separate times and each time looping 1 million times.

## Footnotes